dartsim/src/JointFeatures.cc

```diff
@@ -163,9 +163,151 @@ void JointFeatures::SetJointVelocityCommand(
   {
     joint->setActuatorType(dart::dynamics::Joint::SERVO);
   }
+  // warn about bug https://github.com/dartsim/dart/issues/1583
+  if ((joint->getPositionLowerLimit(_dof) > -1e16 ||
+       joint->getPositionUpperLimit(_dof) < 1e16 ) &&
+      (!std::isfinite(joint->getForceUpperLimit(_dof)) ||
+       !std::isfinite(joint->getForceLowerLimit(_dof))))
+  {
+    static bool informed = false;
+    if (!informed)
+    {
+      ignerr  << "Velocity control does not respect positional limits of "
+              << "joints if these joints do not have an effort limit. Please, "
+              << "set min and max effort for joint [" << joint->getName()
+              << "] to values about -1e6 and 1e6 (or higher if working with "
+              << "heavy links)." << std::endl;
+      informed = true;
+    }
+  }
+
   joint->setCommand(_dof, _value);
 }
 
+/////////////////////////////////////////////////
+void JointFeatures::SetJointMinPositionCommand(
+    const Identity &_id, const std::size_t _dof, const double _value)
+{
+  auto joint = this->ReferenceInterface<JointInfo>(_id)->joint;
+
+  // Take extra care that the value is valid. A nan can cause the DART
+  // constraint solver to fail, which will in turn either cause a crash or
+  // collisions to fail
+  if (std::isnan(_value))
+  {
+    ignerr << "Invalid minimum joint position value [" << _value
+           << "] commanded on joint [" << joint->getName() << " DOF " << _dof
+           << "]. The command will be ignored\n";
+    return;
+  }
+  joint->setPositionLimitEnforced(true);
```

Comment:
`setPositionLimitEnforced` is deprecated:

```suggestion
  joint->setLimitEnforcement(true);
```
Here and below.

Reply:
Actually, we need to support both DART 6.9 and 6.10 so, can you use an `ifdef` to call the correct function like shown in https://github.com/ignitionrobotics/ign-physics/blob/b22bfea7f4c9ccd80b56647ee47fbaf99c7e1acb/test/plugins/DARTDoublePendulum.cc#L77-L82

Reply:
https://github.com/ignitionrobotics/ign-physics/pull/260/commits/4984ceca07eac786607ec3275066b28a17271afd